platform/engineering/request-for-comment/tracking-submissions-from-origin-to-destination.md

```diff
@@ -0,0 +1,399 @@
+
+**Evidence**: Any document uploaded by a user, that vets-api is responsible for transmitting down- or up- stream. Examples could include a medical document, buddy statement, scanned-in filled-out form, digitally filled-out form, and other various documents.
+
+**Ancillary Document**: Any document that has an official VA designated form number, that we fill-out and transmit somewhere, based on information the end-user enters into the frontend. Depending on the form/situation, the end-user may or may not know they filled this form out, or that one was filled out on their behalf. Examples include form 21-4142, form 21-0781, and other forms that we take information and stamp onto a PDF to transmit.
+
+**Documents**: May be used to refer to both evidence and ancillary documents collectively.
+
+**PII**: Personally Identifiable Information. Information with this designation must be protected from view by any unauthorized groups.
+
+**PHI**: Protected Health Information. Any information subject to the privacy protections and patients’ rights provisions of the HIPAA Privacy Rule.
+
+## Motivation
+
+- Provide a way to track unique submissions from system to system
+- Allow for transition time for systems that cannot be updated quickly
+- Can recover from errors or gaps in tracking
+- Should not be considered PII/PHI (but may use PII/PHI in its generation)
+
+## Design
+
+The main design is to generate a unique ID for each submission, regardless of type that can persist between all systems the submission will travel. In order to support systems that cannot or will not be updated to accept and work with this ID, it should be deterministic based on the content of the submission. This allows systems that are aware of the ID to "regenerate" the ID or corelate that system's ID to this unique ID.
+
+A reference implementation for this thought experiment would be a Reporting/BI service would collect information about submissions from each system. This service would attempt to track a submission between these systems. In order to do so, it will need a way to associate records from each system to a unique submission. Using the ideas in this paper, the Reporting service would be able to either:
+
+- Use the generated Submission ID, if it's available from the reporting service
+- Generate the Submission ID from the content of the submission, if it's not available from the reporting service
+
+### Unique and Deterministic
+
+A globally unique ID must be generated for each submission. This ID should be deterministic such that the same submission should result in the same ID. This should be independent of time or place, allowing the ID to be generated or regenerated successfully at any point in the submission's lifecycle.
```

Comment:
Is this true for multiple attempts of the same thing, or sub-components? 

IE would you want something like `vagov526Submission_23344455_attempt_1` to differentiate between multiple attempts at submitting a single thing multiple times with potentially different outcomes?